src/abstractitensornetwork.jl

```diff
@@ -815,9 +819,15 @@ function check_hascommoninds(
     )
   end
   for v in vertices(A)
-    !hascommoninds(siteinds(A, v), siteinds(B, v)) && error(
-      "$(typeof(A)) A and $(typeof(B)) B must share site indices. On vertex $v, A has site indices $(siteinds(A, v)) while B has site indices $(siteinds(B, v)).",
-    )
+    if isempty(siteinds(A, v))
```

Comment:
It seems like `check_hascommoninds` could just call `hascommoninds(siteinds, A, B)` and return `nothing` if it is true and error if it is false.

Reply:
Yes.